front/lib/api/assistant/conversation.ts

```diff
@@ -993,36 +993,38 @@ export async function* editUserMessage(
     };
     return;
   }
-  if (message.mentions.filter((m) => isAgentMention(m)).length > 0) {
-    yield {
-      type: "user_message_error",
-      created: Date.now(),
-      error: {
-        code: "not_allowed",
-        message:
-          "Editing a message that already has agent mentions is not yet supported",
-      },
-    };
-    return;
-  }
 
-  if (
-    !conversation.content[conversation.content.length - 1].some(
-      (m) => m.sId === message.sId
-    ) &&
-    mentions.filter((m) => isAgentMention(m)).length > 0
-  ) {
-    yield {
-      type: "user_message_error",
-      created: Date.now(),
-      error: {
-        code: "edition_unsupported",
-        message:
-          "Adding agent mentions when editing is only supported for the last message of the conversation",
-      },
-    };
-    return;
-  }
+  // Todo these checks are for adding mention, should be ok to edit a message
```

Comment:
First, we need to clean that up :)

Second, this created the assumption that there was no agent message as reply of the user message.

This is not true anymore. Mind describing here in place of that comment what exactly happens when I edit a user message. When it's the last one? When it's not the last one?

Reply:
I cleaned and added a test checking that we are editing the last message. Also added a comment about what happens when editing a message.